sweep/sweeper.go

```diff
@@ -150,42 +118,83 @@ type pendingInput struct {
 	// notifier spend registration.
 	ntfnRegCancel func()
 
-	// state tracks the state this input is in.
-	state inputState
+	// minPublishHeight indicates the minimum block height at which this
+	// input may be (re)published.
+	minPublishHeight int32
 
-	// witnessSizeUpperBound is the upper bound of the number of witness
-	// bytes that this input is going to add to the tx weight.
-	witnessSizeUpperBound int
+	// publishAttempts records the number of attempts that have already been
+	// made to sweep this tx.
+	publishAttempts int
 }
 
 // New returns a new Sweeper instance.
 func New(cfg *UtxoSweeperConfig) *UtxoSweeper {
-	// Calculate dust limit based on the P2WPKH output script of the sweep
-	// txes.
-	dustLimit := txrules.GetDustThreshold(
-		lnwallet.P2WPKHSize,
-		btcutil.Amount(cfg.Estimator.RelayFeePerKW().FeePerKVByte()))
 
 	return &UtxoSweeper{
 		cfg:           cfg,
 		newInputs:     make(chan *sweepInputMessage),
 		spendChan:     make(chan *chainntnfs.SpendDetail),
 		quit:          make(chan struct{}),
 		pendingInputs: make(map[wire.OutPoint]*pendingInput),
-		dustLimit:     dustLimit,
 	}
 }
 
 // Start starts the process of constructing and publish sweep txes.
 func (s *UtxoSweeper) Start() error {
 	if !atomic.CompareAndSwapUint32(&s.started, 0, 1) {
 		return nil
+
 	}
 
 	log.Tracef("Sweeper starting")
 
+	// TODO: Republish last tx? No matter what the result is, the output address should
```

Comment:
On restart, the wallet backend _should_ re-publish all the transactions. However, we also need to ensure we re-register for all the inputs that are spent by our unconfirmed transactions.